crates/wiggle/src/lib.rs

```diff
@@ -613,8 +688,13 @@ impl<'a, T> GuestPtr<'a, [T]> {
     where
         T: GuestTypeTransparent<'a> + Copy + 'a,
     {
-        // bounds check ...
-        let mut self_slice = self.as_slice_mut()?;
+        // Retrieve the slice of memory to copy to, performing the necessary
+        // bounds checks...
+        let mut unsafe_guest_slice = self.as_unsafe_slice_mut()?;
+        // ... SAFETY: since we copy the same regardless of whether the memory
+        // is shared or non-shared, we accept that the data in `&mut [T]` may be
+        // concurrently modified ...
+        let self_slice = unsafe { unsafe_guest_slice.as_mut() };
```

Comment:
We'll need to be careful here because the mere presence of `&mut [T]`, however ephemeral it is, will be UB for Rust that we need to avoid. Basically the raw-pointer-ness needs to get plumbed all the way down to the `std::ptr::copy`.

Also, can you be sure to tag this `std::ptr::copy` in the threads issue as "please audit at some point" along the lines of the other copies already in `runtime/src/instances.rs`